chain/watcher_test.go

```diff
@@ -80,14 +87,15 @@ func TestChainHeadIndexer(t *testing.T) {
 
 	cids := bhs.Cids()
 	rounds := bhs.Rounds()
-	chainHead, err := node.ChainHead(ctx)
-	require.NoError(t, err, "ChainHead")
-
-	tipSetKeys := []string{chainHead.Key().String()}
 
 	err = idx.index(ctx, nh)
 	require.NoError(t, err, "index")
 
+	// TODO NewTipSetIndexer runs its processors in their own go routines (started when TipSet() is called)
+	// this causes this test to behave nondeterministicly so we sleep here to ensure all async jobs
+	// have completed before asserting results
+	time.Sleep(time.Second * 3)
```

Comment:
This test is still useful and I have done my best to keep it working without changing implementation logic (Trying to keep this change as mechanical as possible). If we are alright with this fix/hack for now I can file an issue to fix this as a follow on :+1: :-1:

Reply:
Happy to keep it like this